### PR TITLE
github: save build log

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
save build log
split building samples using matrix to distribute load even move generation of PR commit message to separate action unify action for PR and commit

### CODE REVIEW

**Review by ktaborowski — APPROVED** [review]

File: .github/workflows/on_PR.yml [hunk 1]
```diff
@@ -54,49 +50,30 @@ jobs:
...
       - name: Upload build artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: sample-${{ matrix.samples_to_test }}
+          name: sample-${{matrix.subset}}_of_${{ strategy.job-total }}
           path: |
+            twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/merged.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.elf
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/dfu_application.zip
```
Comment (totyz) [inline .github/workflows/on_PR.yml:64]: Please add also or instead of dfu_application.zip, app_update.bin

File: .github/workflows/on_commit.yml [hunk 2]
```diff
@@ -30,47 +50,30 @@ jobs:
...
       - name: Upload build artifacts
         uses: actions/upload-artifact@v3
         with:
-          name: sample-${{ matrix.samples_to_test }}
+          name: sample-${{matrix.subset}}_of_${{ strategy.job-total }}
           path: |
+            twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/merged.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.elf
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/zephyr.hex
             twister-out/**/sidewalk/samples/**/*sidewalk*/zephyr/dfu_application.zip
```
Comment (totyz) [inline .github/workflows/on_commit.yml:64]: Please add also or instead of dfu_application.zip, app_update.bin

**Review by totyz — CHANGES_REQUESTED** [review]

File: .github/workflows/on_PR.yml (hunk 1 above)
Comment (RobertGalatNordic) [reply .github/workflows/on_PR.yml:65]: done

File: .github/workflows/on_commit.yml (hunk 2 above)
Comment (RobertGalatNordic) [reply .github/workflows/on_commit.yml:65]: done

**Review by totyz — APPROVED** [review]

**Pull request merged**